lib/rspec/mocks/error_generator.rb

```diff
@@ -62,6 +62,18 @@ def raise_unexpected_message_args_error(expectation, *args)
       end
 
       # @private
+      def raise_unexpected_message_with_args_error(expectation, *args)
+        expected_args = format_args(*expectation.expected_args)
+        actual_args = format_unexpected_with_args(*args)
+        diff = diff_message(expectation.expected_args, args)
+
+        message = default_error_message(expectation, expected_args, actual_args)
+        message << "\nDiff:#{diff}" unless diff.empty?
+
+        __raise message
+      end
```

Comment:
This method is nearly identical to the one above it (`raise_unexpected_message_args_error`).  From the names, it's not clear to me what the difference is intended to be.  Implementation-wise, the only difference seems to be the way the args are treated (one has `actual_args = format_received_args(*args)` and the other has `actual_args = format_unexpected_with_args(*args)`).  A few thoughts:

* Can we combine these into one method?  I think the difference is that the existing one expects `args` to be a list of the args for a single method call, where as for this method, `args` is multiple lists of args from multiple method calls...is that right?  Given that, I think we could standardize on _one_ method, where the existing callers of the old method would wrap the args in an array (`[args]`) in order to cause them to be treated as the args of a single method call.
* There's a lot of splatting and un-splatting going on here.  I know you're just following the patterns you see in this file but I think it makes sense to move away from that for new code like this -- I believe there's a cost associated with splatting and un-splatting that's not there when you just pass an array as-is.